### PR TITLE
Upgrade Django 3.2

### PR DESCRIPTION
**Pull request by hthungdot**

### CODE REVIEW

**Comment by NaviRocker** [comment]
Hi @hthungdot, I noticed that you are new to contributions. That's a great start! However, I would like to offer some advice. Before submitting a pull request, please claim the issue and refer to it in the pull request's description. Additionally, provide some information on what exactly you have done. This will make it easier for the reviewer to assess. Once again, best of luck with your upcoming contributions. Cheers!

File: src/pretix/api/serializers/order.py [hunk 1]
```diff
@@ -292,8 +297,11 @@ def to_representation(self, instance: OrderPosition):
                 self.context['vars_images'] = get_images(self.context['request'].event)
 
             for k, f in self.context['vars'].items():
-                res[k] = f['evaluate'](instance, instance.order, ev)
-
+                try:
+                    res[k] = f['evaluate'](instance, instance.order, ev)
+                except:
```
Comment (hongquan) [inline src/pretix/api/serializers/order.py:302]: Please use more specfic error class with `except`. Don't use bare `except` or `except Exception`.

File: src/pretix/api/serializers/order.py [hunk 2]
```diff
@@ -309,10 +317,20 @@ def to_representation(self, instance: OrderPosition):
 
             for k, f in self.context['vars_images'].items():
                 if 'etag' in f:
-                    has_image = etag = f['etag'](instance, instance.order, ev)
-                else:
-                    has_image = f['etag'](instance, instance.order, ev)
-                    etag = None
+                    try:
+                        has_image = etag = f['etag'](instance, instance.order, ev)
+                    except:
```
Comment (hongquan) [inline src/pretix/api/serializers/order.py:322]: Same here for `except`.

**Review by hongquan — CHANGES_REQUESTED** [review]

**Comment by eMBee** [comment]
how did you install django 3.2 to test this? the requirements files have not been updated to reflect the new versions needed. see src/requirements/*.txt

**Comment by hthungdot** [comment]
@eMBee This pull request is not finished, but I committed what I have done so far. I hope someone else can finish it.

**Comment by eMBee** [comment]
> @eMBee This pull request is not finished, but I committed what I have done so far. I hope someone else can finish it.

ok, no worries. could you please outline which steps you took to get this far? i think it would help me and others to continue the work if we know how you created the current patch.

File: src/pretix/presale/urls.py [hunk 3]
```diff
@@ -1,8 +1,9 @@
-from django.conf.urls import include, url
+from django.conf.urls import include, re_path
 from django.views.decorators.csrf import csrf_exempt
 
 import pretix.presale.views.cart
 import pretix.presale.views.checkout
+import pretix.presale.views.customer
```
Comment (hongquan) [inline src/pretix/presale/urls.py:6]: This "customer" module doesn't exist.
Don't know how you come up with the name of the views imported from it.

**Review by hongquan — CHANGES_REQUESTED** [review]

**Pull request closed without merging**